### PR TITLE
Use generators for any and all calls and comprehensions

### PR DESCRIPTION
**Pull request by tweakimp**
I just made some small improvements:
In `any` and `all` calls, generators are faster than lists because the loop can break and short cut if the first `True` (in `any`) or `False` (in `all`) is found.
Also some cleaner code with comprehensions :)

### CODE REVIEW

**Comment by dor2727** [comment]
shouldn't you, following the same logic, in the 2 changes in "active_projects/eop/reusables/histograms.py", change the square brackets into parentheses? So that it would return a generator?

**Comment by tweakimp** [comment]
no, because i dont know where they are used. maybe the function that calls them expects them to be lists.

if i know the caller just iterates over the results, i could turn the called functions into generators with the yield statement, but that requires deeper analysis of the code

File: active_projects/eop/reusables/histograms.py [hunk 1]
```diff
@@ -82,10 +78,7 @@ def generate_points(self):
             self.remove(submob)
 
         def empty_string_array(n):
-            arr = []
-            for i in range(n):
-                arr.append("")
-            return arr
+            return ["" for _ in range(n)]
```
Comment (miggazElquez) [inline active_projects/eop/reusables/histograms.py]: Why not ``[""] * n`` ?

**Comment by TonyCrane** [comment]
Seems outdated, so I'm closing this.

**Pull request closed without merging**